pgmpy/factors/Factor.py

```diff
@@ -365,6 +378,16 @@ def reduce(self, values, inplace=True):
         >>> phi.values
         array([0., 1.])
         """
+
+        if isinstance(values, str):
+            raise TypeError("values: Expected type list or array-like, got type str")
+
+        if any(isinstance(value, str) for value in values):
+            raise TypeError("values: must contain tuples or array-like elements of the form (type str, type int)")
+
+        if not all(isinstance(var, str) and isinstance(state, int) for var,state in values):
```

Comment:
@yashu-seth Since the error message is the same in both this case and the one just above this. You can merge the conditions with an `or` .